tests/unit/clients/memcache_tests.py

```diff
@@ -10,11 +10,87 @@
 else:
     del pymemcache
 
+from prometheus_client import REGISTRY
 from baseplate.lib.config import ConfigurationError
 from baseplate.clients.memcache import pool_from_config
+from baseplate.clients.memcache import MonitoredMemcacheConnection
 from baseplate.clients.memcache import lib as memcache_lib
 
 
+class PrometheusInstrumentationTests(unittest.TestCase):
+    # test the @_prom_instrument decorator by calling one of the decorated functions
+    def test_prometheus_decorator(self):
+        # stub out the server span, each call to MonitoredMemcacheConnection
+        # will create a new span - but we don't use that for prom
+        span = mock.Mock()
+        span_inner = mock.Mock()
+        span_inner.__enter__ = mock.Mock(return_value=mock.Mock())
+        span_inner.__exit__ = mock.Mock(return_value=None)
+        span.make_child = lambda trace_name: span_inner
+
+        # stub out the client, we only use the server name
+        pooled_client = mock.Mock()
+        pooled_client.server = "server name"
+
+        prom_labels = {
+            "memcached_address": "server name",
+            "memcached_command": "incr",
+        }
+
+        mmc = MonitoredMemcacheConnection("test", span, pooled_client)
+        mmc.incr("blah", 1)
+
+        assert (
+            REGISTRY.get_sample_value(
+                "memcached_client_active_requests",
+                prom_labels,
+            )
+            == 0
+        )
+        assert (
+            REGISTRY.get_sample_value(
+                "memcached_client_requests_total",
+                {
+                    **prom_labels,
+                    "memcached_success": "true",
+                },
+            )
+            == 1
+        )
+        assert (
+            REGISTRY.get_sample_value(
+                "memcached_client_latency_seconds_bucket",
+                {
+                    **prom_labels,
+                    "memcached_success": "true",
+                    "le": "+Inf",
+                },
+            )
+            == 1
+        )
+
+        class FailClient(mock.Mock):
+            server = "hello"
+
+            def incr(self, _a, _b, noreply=None):
+                raise TypeError
+
+        fail_mmc = MonitoredMemcacheConnection("test", span, FailClient())
```

Comment:
Can your failure test case actually be a different test case. As written if the success case fails so does this one.

Reply:
done